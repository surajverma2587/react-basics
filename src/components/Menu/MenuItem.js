import React from 'react';
import PropTypes from 'prop-types';

const MenuItem = (/* props object */) => {
  /* bind the data from the props object to the html template */
  /* validate the props and render the component IF valid ELSE null */
  /* replace the bottom return statement with the populated html template */
  return (
    <div>
      MenuItem Component
    </div>
  );
};

/* display name for the component */
MenuItem.displayName = 'MenuItem';

MenuItem.propTypes = {
  /* add Proptypes here */
};

export default MenuItem;