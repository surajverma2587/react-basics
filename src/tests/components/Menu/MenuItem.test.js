import React from 'react';
import { shallow } from 'enzyme';

import MenuItem from '../../../components/Menu/MenuItem';

const props = {
  id: '1',
  title: 'Some title',
  description: 'Some description',
  price: 10,
  onClick: jest.fn()
};

describe('MenuItem Component', () => {
  describe('when props are valid', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<MenuItem {...props} />);
    });
    
    it('should render snapshot of the component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should render the component', () => {
      const menuItem = wrapper.find('.menu_item');
      expect(menuItem.exists()).toBe(true);
    });
  
    it('should render the title', () => {
      const title = wrapper.find('.menu_item--title');
      expect(title.text()).toEqual(props.title);
    });
  
    it('should render the description', () => {
      const description = wrapper.find('.menu_item--description');
      expect(description.text()).toEqual(props.description);
    });
  
    it('should render the price', () => {
      const price = wrapper.find('.menu_item--price');
      expect(price.text()).toEqual(`Â£${props.price}`);
    });
  
    it('should render the add button', () => {
      const button = wrapper.find('button');
      expect(button.prop('id')).toEqual(props.id);
      expect(button.prop('onClick')).toEqual(props.onClick);
    });
  });
  
  describe('when props are invalid', () => {
    it('should not render the component when props are not present', () => {
      const wrapper = shallow(<MenuItem />);
      expect(wrapper.find('.menu_item').exists()).toBe(false);
    });
  
    it('should not render the component when any prop is invalid', () => {
      const invalidProps = [
        { ...props, id: null },
        { ...props, title: null },
        { ...props, description: null },
        { ...props, price: null },
        { ...props, onClick: null },
      ];
  
      invalidProps.forEach(invalidProp => {
        const wrapper = shallow(<MenuItem {...invalidProp} />);
        expect(wrapper.find('.menu_item').exists()).toBe(false);
      });
    });
  });
});
