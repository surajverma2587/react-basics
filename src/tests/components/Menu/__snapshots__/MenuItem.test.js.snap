// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem Component when props are valid should render snapshot of the component 1`] = `
<div
  className="menu_item"
>
  <div
    className="row"
  >
    <div
      className="col-sm-10 menu_item--title"
    >
      <span>
        Item title
      </span>
    </div>
    <div
      className="col-sm-1 menu_item--price"
    >
      <span>
        Â£
        5.99
      </span>
    </div>
    <div
      className="col-sm-1"
    >
      <button
        className="btn btn-success"
        id="I1"
        onClick={[MockFunction]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-sm-12 menu_item--description"
    >
      <span>
        Item description
      </span>
    </div>
  </div>
</div>
`;
